src/main/java/de/thomas_oster/liblasercut/drivers/ThunderLaserProperty.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * This file is part of LibLaserCut.
+ * Copyright (C) 2011 - 2014 Thomas Oster <mail@thomas-oster.de>
+ * Copyright (c) 2018 Klaus Kämpf <kkaempf@suse.de>
+ *
+ * LibLaserCut is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * LibLaserCut is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibLaserCut. If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+package de.thomas_oster.liblasercut.drivers;
+
+import de.thomas_oster.liblasercut.FloatPowerSpeedFocusProperty;
+import java.util.Arrays;
+
+/**
+ *
+ * @author kkaempf
+ *
+ * It's all pretty ugly. This should rather extend PowerSpeedFocusFrequencyProperty.
+ * But this is incompatible with RasterizableJobPart(), which is expecting a FloatPowerSpeedFocusProperty :-/
```

Comment:
I would need help here ...

Reply:
have a look in my ruida-help branch, the RasterPart stuff should now be general. I haven't tested the code, so expect some bugs (or some accidental breakage to other drivers such as the Epilog driver)

Reply:
ruida-help indeed introduced two breakages, one in "make" and one runtime. See https://github.com/mgmax/LibLaserCut/pull/1 :wink:

Reply:
Done. cherry-picked your Property fixes from `ruida-help`

Reply:
The bugs are fixed (see https://github.com/kkaempf/LibLaserCut/pull/1). If you want to switch to integer-based laser properties (PowerSpeedFocusFrequency uses Integer) then you can do that now. Otherwise the comment should be dropped.